src/main/java/kernel/serial_ports/RXTXPortDriver.java

```diff
@@ -0,0 +1,89 @@
+package kernel.serial_ports;
+
+import gnu.io.CommPortIdentifier;
+import kernel.serial_ports.comm_port_wrapper.CommPortWrapper;
+
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+
+/**
+ * Driver responsible for managing serial port names
+ */
+public class RXTXPortDriver implements PortDriver {
+    private final String applicationName;
+    private final CommPortWrapper portWrapper;
+
+    public RXTXPortDriver(String applicationName, CommPortWrapper portWrapper){
+        this.applicationName = applicationName;
+        this.portWrapper = portWrapper;
+    }
+
+    @Override public String getApplicationName(){
+        return this.applicationName;
+    }
+
+    @Override public List<String> getSerialPortNames(){
+        Enumeration portIdentifiers = this.portWrapper.getPortIdentifiers();
+        List<CommPortIdentifier> identifierList = castPortIdentsToList(
+            portIdentifiers);
+
+        List<CommPortIdentifier> serialPortList = filterForSerialPorts(
+                identifierList
+        );
+
+        return getPortNamesForList(serialPortList);
+    }
+
+    @Override public SerialPort getPortByName(String portName) {
+        return new RXTXPortWrapper(portName);
+    }
+
+    private ArrayList<CommPortIdentifier> castPortIdentsToList(
+            Enumeration commPortIdentifiers
+    ){
+        ArrayList<CommPortIdentifier> portIdentifierList = new ArrayList<>();
```

Comment:
Change the constructor from ``ArrayList`` to ``Vector``. ``Vector``s behave much more predictably with undefined array sizes.